sklearn/compose/_column_transformer.py

```diff
@@ -511,6 +513,13 @@ def _hstack(self, Xs):
             return np.hstack(Xs)
 
 
+def _check_X(X):
+    """Use check_array only on lists and other non-array-likes / sparse"""
+    if hasattr(X, '__array__') or sparse.issparse(X):
+        return X
+    return check_array(X)
```

Comment:
I think we still need to allow missing values here? (`force_all_finite='allow-nan'`)